pkg/download/buffer.go

```diff
@@ -37,12 +37,30 @@ func (t *Target) Basename() string {
 	return filepath.Base(t.Dest)
 }
 
-func getBufferMode(client *client.HTTPClient) Mode {
+func getBufferMode(opts Options) Mode {
+	client := client.NewHTTPClient(opts.Client)
 	return &BufferMode{
-		Client: client,
+		Client:  client,
+		Options: opts,
 	}
 }
 
+func (m *BufferMode) maxChunks() int {
+	maxChunks := m.MaxChunks
+	if maxChunks == 0 {
+		return runtime.GOMAXPROCS(0) * 4
+	}
+	return maxChunks
+}
+
+func (m *BufferMode) minChunkSize() int64 {
+	minChunkSize := m.MinChunkSize
+	if minChunkSize == 0 {
+		return 16 * 1024 * 1024
```

Comment:
Should this be a constant? Does this get optimized out in compilation?

No change needed here.